### PR TITLE
Feature ID 161 - Get current location in real time

### PR DESCRIPTION
**Pull request by fabianromeroclaros**
## Get user location in real time implemented.
- The user location is displayed and shown when the app is opened and when user pressed the button.
- Also when user pressed the button the camera on the mapa is centered to the location.
- Access to location device permissions are shown to user before trying to get the location and verified every time is needed.

### For testing, you have to set your location in the emulator or you could test it in a real device (Recommended).

_How to set location in emulator?_
https://stackoverflow.com/questions/2279647/how-to-emulate-gps-location-in-the-android-emulator

**Acceptance criteria**

- [x] Get current location of the user every time it is updated in real time.
- [x] Center the map to the current location when get location button is pressed
- [x] Ask for permissions to user to access to the location before trying to get it
- [x] The get location button should be an icon button in the left down corner
- [x] The current location should be in the map by an icon displayed on it
- [x] The current location should be gotten when opening the app and when pressing the icon button

### CODE REVIEW

File: app/src/main/java/com/isc/hermes/controller/CurrentLocationController.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
+package com.isc.hermes.controller;
+
+import android.annotation.SuppressLint;
+import android.widget.ImageButton;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/CurrentLocationController.java:8]: Delete the spaces that you do not get to use.

File: app/src/main/java/com/isc/hermes/controller/LocationPermissionsController.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package com.isc.hermes.controller;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/LocationPermissionsController.java:9]: It also removes unused spaces.

File: app/src/main/java/com/isc/hermes/utils/LocationListeningCallback.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package com.isc.hermes.utils;
+
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/utils/LocationListeningCallback.java:2]: Delete the spaces that you do not get to use.

**Review by VillalobosGarciaGaby — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/controller/CurrentLocationController.java (hunk 1 above)
Comment (victor0villca) [reply app/src/main/java/com/isc/hermes/controller/CurrentLocationController.java]: @VillalobosGarciaGaby  This recommendation is already satisfied.

File: app/src/main/java/com/isc/hermes/controller/LocationPermissionsController.java (hunk 2 above)
Comment (victor0villca) [reply app/src/main/java/com/isc/hermes/controller/LocationPermissionsController.java]: @VillalobosGarciaGaby  This recommendation is already satisfied.

File: app/src/main/java/com/isc/hermes/utils/LocationListeningCallback.java (hunk 3 above)
Comment (victor0villca) [reply app/src/main/java/com/isc/hermes/utils/LocationListeningCallback.java]: @VillalobosGarciaGaby  This recommendation is already satisfied.

**Review by saletG08 — APPROVED** [review]
Everything is correct, approved

**Review by santiCaballeroM — APPROVED** [review]
Approved

File: app/src/main/java/com/isc/hermes/controller/LocationPermissionsController.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        Toast.makeText(
+                activity, "Need permissions for access to location", Toast.LENGTH_SHORT
+        ).show();
+    }
+
+    /**
+     * Method called when permissions are granted.
+     */
+    @Override
+    public void onPermissionResult(boolean granted) {
```
[the remark this answers is not in the document]
Comment (fabianromeroclaros) [reply app/src/main/java/com/isc/hermes/controller/LocationPermissionsController.java]: Done

File: app/src/main/java/com/isc/hermes/model/LocationPermissionsModel.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.isc.hermes.model;
+
+/**
+ * The LocationPermissionsModel class represents the model for location permissions.
+ * It stores the granted status of location permissions.
+ */
+public class LocationPermissionsModel {
+    boolean granted;
```
[the remark this answers is not in the document]
Comment (fabianromeroclaros) [reply app/src/main/java/com/isc/hermes/model/LocationPermissionsModel.java]: Done

File: app/src/main/java/com/isc/hermes/utils/LocationListeningCallback.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    /**
+     * Called when the location engine successfully retrieves a location update.
+     * Updates the CurrentLocationModel with the latitude and longitude of the received location.
+     *
+     * @param result The LocationEngineResult containing the last received location.
+     */
+    @Override
+    public void onSuccess(LocationEngineResult result) {
+        AppCompatActivity activity = activityWeakReference.get();
+        if (activity != null) {
```
[the remark this answers is not in the document]
Comment (fabianromeroclaros) [reply app/src/main/java/com/isc/hermes/utils/LocationListeningCallback.java]: Done

**Pull request merged**